src/app/identity.rs

```diff
@@ -164,63 +185,95 @@ where
                         }
                         Err(e) => {
                             error!("Failed to read authentication token: {}", e);
                             Inner::Waiting(self.config.refresh(self.expiry))
                         }
                     }
                 }
-                Inner::Pending(ref mut p) => match p.poll() {
-                    Ok(Async::NotReady) => return Ok(Async::NotReady),
-                    Ok(Async::Ready(rsp)) => {
-                        let api::CertifyResponse {
-                            leaf_certificate,
-                            intermediate_certificates,
-                            valid_until,
-                        } = rsp.into_inner();
-
-                        match valid_until.and_then(|d| Result::<SystemTime, Duration>::from(d).ok())
-                        {
-                            None => error!(
-                                "Identity service did not specify a ceritificate expiration."
-                            ),
-                            Some(expiry) => {
-                                let key = self.config.key.clone();
-                                let crt = Crt::new(
-                                    self.config.local_name.clone(),
-                                    leaf_certificate,
-                                    intermediate_certificates,
-                                    expiry,
-                                );
-
-                                match self.config.trust_anchors.certify(key, crt) {
-                                    Err(e) => {
-                                        error!("Received invalid ceritficate: {}", e);
-                                    }
-                                    Ok(crt_key) => {
-                                        if self.crt_key.store(Some(crt_key)).is_err() {
-                                            // If we can't store a value, than all observations
-                                            // have been dropped and we can stop refreshing.
-                                            return Ok(Async::Ready(()));
+                Inner::Pending(ref mut p) => {
+                    trace!("daemon pending certification");
+                    match p.poll() {
+                        Ok(Async::NotReady) => return Ok(Async::NotReady),
+                        Ok(Async::Ready(rsp)) => {
+                            let api::CertifyResponse {
+                                leaf_certificate,
+                                intermediate_certificates,
+                                valid_until,
+                            } = rsp.into_inner();
+
+                            match valid_until
+                                .and_then(|d| Result::<SystemTime, Duration>::from(d).ok())
+                            {
+                                None => error!(
+                                    "Identity service did not specify a ceritificate expiration."
```

Comment:
Typo: "ceritificate"
```suggestion
                                    "Identity service did not specify a certificate expiration."
```